paging-runtime-composeui/src/commonMain/kotlin/app/cash/paging/LazyPagingItems.kt

```diff
@@ -0,0 +1,320 @@
+   * trigger any loads necessary to fulfill prefetchDistance.
+   *
+   * @see peek
+   */
+  operator fun get(index: Int): T? {
+    pagingDataDiffer[index] // this registers the value load
+    return itemSnapshotList[index]
+  }
+
+  /**
+   * Returns the presented item at the specified position, without notifying Paging of the item
+   * access that would normally trigger page loads.
+   *
+   * @param index Index of the presented item to return, including placeholders.
+   * @return The presented item at position [index], `null` if it is a placeholder
+   */
+  fun peek(index: Int): T? {
+    return itemSnapshotList[index]
+  }
+
+  /**
+   * Retry any failed load requests that would result in a [LoadState.Error] update to this
+   * [LazyPagingItems].
+   *
+   * Unlike [refresh], this does not invalidate [PagingSource], it only retries failed loads
+   * within the same generation of [PagingData].
+   *
+   * [LoadState.Error] can be generated from two types of load requests:
+   *  * [PagingSource.load] returning [PagingSource.LoadResult.Error]
+   *  * [RemoteMediator.load] returning [RemoteMediator.MediatorResult.Error]
+   */
+  fun retry() {
+    pagingDataDiffer.retry()
+  }
+
+  /**
+   * Refresh the data presented by this [LazyPagingItems].
+   *
+   * [refresh] triggers the creation of a new [PagingData] with a new instance of [PagingSource]
+   * to represent an updated snapshot of the backing dataset. If a [RemoteMediator] is set,
+   * calling [refresh] will also trigger a call to [RemoteMediator.load] with [LoadType] [REFRESH]
+   * to allow [RemoteMediator] to check for updates to the dataset backing [PagingSource].
+   *
+   * Note: This API is intended for UI-driven refresh signals, such as swipe-to-refresh.
+   * Invalidation due repository-layer signals, such as DB-updates, should instead use
+   * [PagingSource.invalidate].
+   *
+   * @see PagingSource.invalidate
+   */
+  fun refresh() {
+    pagingDataDiffer.refresh()
+  }
+
+  /**
+   * A [CombinedLoadStates] object which represents the current loading state.
+   */
+  var loadState: CombinedLoadStates by mutableStateOf(
+    CombinedLoadStates(
+      refresh = InitialLoadStates.refresh,
+      prepend = InitialLoadStates.prepend,
+      append = InitialLoadStates.append,
+      source = InitialLoadStates
+    )
+  )
+    private set
+
+  internal suspend fun collectLoadState() {
+    pagingDataDiffer.loadStateFlow.collect {
+      loadState = it
+    }
+  }
+
+  internal suspend fun collectPagingData() {
+    flow.collectLatest {
+      pagingDataDiffer.collectFrom(it)
+    }
+  }
+
+  private companion object {
+    init {
+      /**
+       * Implements the Logger interface from different build targets
+       * and injects it into the LOGGER global var stored within Pager.
+       *
+       * Checks for null LOGGER because other runtime
+       * entry points to paging can also inject a Logger
+       */
+      LOGGER = LOGGER ?: createLogger()
```

Comment:
Here's the one change I made in all of this source, which injects a target specific logger instead of the hard-coded android one used in androidx source.